packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -153,6 +159,32 @@ export class QueuedRequestController extends BaseController<
       `${controllerName}:enqueueRequest`,
       this.enqueueRequest.bind(this),
     );
+
+    this.messagingSystem.subscribe(
+      SelectedNetworkControllerEventTypes.stateChange,
+      (_, patch) => {
+        patch.forEach(({ path, value }) => {
+          if (path[0] === 'domains') {
+            this.#flushQueueForOrigin(value);
+          }
```

Comment:
wouldn't `path[1]` be the origin? I may be misunderstanding how the patches are generated

Reply:
Ill double check this, thanks!

Reply:
oof yeah this was wrong. Not sure how I got this wrong... hm.